Tests/ApolloTests/Cache/ReadWriteFromStoreTests.swift

```diff
@@ -609,4 +609,155 @@ class ReadWriteFromStoreTests: XCTestCase, CacheDependentTesting, StoreLoading {
       }
     }
   }
+
+  func testRemoveObjectsByPattern() throws {
```

Comment:
This is a long test, if you have suggestions on how to trim it down please say so.

Reply:
I would suggest breaking each piece up after `mergeRecordsIntoCache` into convenience methods. This will also make it less horrendous to write more tests on this in the future if needed.

Reply:
Yeah helper functions would be nice, but in general tests are often longer than is acceptable for production code functions. I'm okay with this as is.

Reply:
`func test_removeObjectsMatchingPattern_givenMultipleObjectsMatchingPattern_deletesAllMatchingRecords()`

Reply:
> but in general tests are often longer than is acceptable for production code functions. I'm okay with this as is.

I generally agree with this but there's a number of things here that are getting repeated both within the test and in another test. That's usually the point at which I start looking to consolidate functions for readability.

Reply:
I'll look for ways to consolidate functions in another PR which can then be applied to that whole test file.